numpyro/util.py

```diff
@@ -378,10 +386,15 @@ def loop_fn(collection):
 
         def loop_fn(collection):
             return fori_loop(
-                0, upper, _body_fn_pbar, (init_val, collection, start_idx, thinning)
+                0,
+                upper,
+                _body_fn_pbar,
+                ((init_val, collection, start_idx, thinning), -1),  # -1 for chain id
             )
 
-        last_val, collection, _, _ = maybe_jit(loop_fn, donate_argnums=0)(collection)
+        (last_val, collection, _, _), _ = maybe_jit(loop_fn, donate_argnums=0)(
```

Comment:
no need to change this if we dont return chain in loop_fn

Reply:
done